src/gsy_e/models/strategy/energy_parameters/heatpump/pcm_tank/pcm_tank.py

```diff
@@ -0,0 +1,504 @@
+            slot_length=GlobalConfig.slot_length, mass_flow_rate_kg_s=MASS_FLOW_RATE
+        )
+        self._pcm_discharge_model = PCMDischargeModel(
+            slot_length=GlobalConfig.slot_length, mass_flow_rate_kg_s=MASS_FLOW_RATE
+        )
+
+    def serialize(self):
+        """Return serializable dict of class parameters."""
+        return {
+            "initial_temp_C": self._initial_temp_C,
+            "min_storage_temp_C": self.min_storage_temp_C,
+            "max_storage_temp_C": self.max_storage_temp_C,
+            "max_capacity_kWh": self._max_capacity_kWh,
+        }
+
+    def update_consumed_energy(self, energy: float, time_slot: DateTime) -> None:
+        """Add provided energy value to the value in consumed_energy for the provided time_slot."""
+        self._consumed_energy[time_slot] = self._consumed_energy.get(time_slot, 0.0) + energy
+
+    def get_consumed_energy(self, time_slot: DateTime):
+        """Return consumed_energy for the provided time_slot"""
+        return self._consumed_energy.get(time_slot, 0.0)
+
+    def init_storage_temps(self, time_slot: DateTime):
+        """
+        Initiate the storage temperatures with the initial temperature of the storge if the
+        _storage_temp_C dict is empty
+        """
+        if len(self._storage_temp_C.keys()) == 0:
+            self._storage_temp_C[time_slot] = [
+                self._initial_temp_C for _ in range(NUMBER_OF_PCM_ELEMENTS)
+            ]
+
+    def is_time_slot_available(self, time_slot: DateTime) -> bool:
+        """Return True if the provided time_slot is part of the _storage_temp_C dict."""
+        return time_slot in self._storage_temp_C
+
+    def set_storage_temp_C(self, storage_temp_C: list[float], time_slot: DateTime) -> None:
+        """Set storage_temp_C for specific time_slot."""
+        self._storage_temp_C[time_slot] = storage_temp_C
+
+    def get_storage_temp_C(self, time_slot: DateTime) -> Optional[list]:
+        """Get storage_temp_C for specific time_slot."""
+        return self._storage_temp_C.get(time_slot, None)
+
+    def set_soc_after_charging(self, time_slot: DateTime):
+        """Calculate and set SOC level after charging."""
+        self._soc[time_slot] = self._pcm_charge_model.get_soc(self._storage_temp_C[time_slot])
+
+    def set_soc_after_discharging(self, time_slot: DateTime):
+        """Calculate and set SOC level after discharging."""
+        self._soc[time_slot] = self._pcm_discharge_model.get_soc(self._storage_temp_C[time_slot])
+
+    def get_soc(self, time_slot: DateTime) -> float:
+        """Return SOC level."""
+        return self._soc[time_slot]
+
+    def get_htf_temp_C(self, time_slot: DateTime) -> Optional[float]:
+        """Return mean temperature of the heat transfer fluid"""
+        storage_temps = self.get_storage_temp_C(time_slot)
+        if storage_temps is None:
+            return None
+        return mean(storage_temps[0:-2:2])
```

Comment:
For readability, why not distinguishing the HTF and the PCM temps in different lists ?

Reply:
I am not very sure about this any more. Wenn calling the (dis-)charge model we will need to combine these lists again into ne (in order to not having to change in the greybox model). And then we need to return two lists again from the models side in order to dave them here in the state separately. Do you think it is worth it?